src/libraries/System.Speech/src/AudioFormat/AudioFormatConverter.cs

```diff
@@ -0,0 +1,300 @@
+                waveEx.nAvgBytesPerSec = waveEx.nSamplesPerSec * waveEx.nBlockAlign;
+            }
+            else if (eFormat == StreamFormat.TrueSpeech_8kHz1BitMono)
+            {
+                waveEx.wFormatTag = (ushort)WaveFormatId.TrueSpeech;
+                waveEx.nChannels = 1;
+                waveEx.nSamplesPerSec = 8000;
+                waveEx.nAvgBytesPerSec = 1067;
+                waveEx.nBlockAlign = 32;
+                waveEx.wBitsPerSample = 1;
+                waveEx.cbSize = 32;
+                extra = new byte[32];
+                extra[0] = 1;
+                extra[2] = 0xF0;
+            }
+            else if ((eFormat >= StreamFormat.CCITT_ALaw_8kHzMono) && (eFormat <= StreamFormat.CCITT_ALaw_44kHzStereo))
+            {
+                uint index = (uint)(eFormat - StreamFormat.CCITT_ALaw_8kHzMono);
+                uint dwKHZ = index / 2;
+                uint[] adwKHZ = { 8000, 11025, 22050, 44100 };
+                bool isStereo = (index & 0x1) != 0;
+                waveEx.wFormatTag = (ushort)WaveFormatId.Alaw;
+                waveEx.nChannels = waveEx.nBlockAlign = (ushort)(isStereo ? 2 : 1);
+                waveEx.nSamplesPerSec = adwKHZ[dwKHZ];
+                waveEx.wBitsPerSample = 8;
+                waveEx.nAvgBytesPerSec = waveEx.nSamplesPerSec * waveEx.nBlockAlign;
+            }
+            else if ((eFormat >= StreamFormat.CCITT_uLaw_8kHzMono) &&
+                (eFormat <= StreamFormat.CCITT_uLaw_44kHzStereo))
+            {
+                uint index = (uint)(eFormat - StreamFormat.CCITT_uLaw_8kHzMono);
+                uint dwKHZ = index / 2;
+                uint[] adwKHZ = new uint[] { 8000, 11025, 22050, 44100 };
+                bool isStereo = (index & 0x1) != 0;
+                waveEx.wFormatTag = (ushort)WaveFormatId.Mulaw;
+                waveEx.nChannels = waveEx.nBlockAlign = (ushort)(isStereo ? 2 : 1);
+                waveEx.nSamplesPerSec = adwKHZ[dwKHZ];
+                waveEx.wBitsPerSample = 8;
+                waveEx.nAvgBytesPerSec = waveEx.nSamplesPerSec * waveEx.nBlockAlign;
+            }
+            else if ((eFormat >= StreamFormat.ADPCM_8kHzMono) &&
+                (eFormat <= StreamFormat.ADPCM_44kHzStereo))
+            {
+                //--- Some of these values seem odd. We used what the codec told us.
+                uint[] adwKHZ = new uint[] { 8000, 11025, 22050, 44100 };
+                uint[] BytesPerSec = new uint[] { 4096, 8192, 5644, 11289, 11155, 22311, 22179, 44359 };
+                uint[] BlockAlign = new uint[] { 256, 256, 512, 1024 };
+                byte[] Extra811 = new byte[32]
+            {
+                0xF4, 0x01, 0x07, 0x00, 0x00, 0x01, 0x00, 0x00,
+                0x00, 0x02, 0x00, 0xFF, 0x00, 0x00, 0x00, 0x00,
+                0xC0, 0x00, 0x40, 0x00, 0xF0, 0x00, 0x00, 0x00,
+                0xCC, 0x01, 0x30, 0xFF, 0x88, 0x01, 0x18, 0xFF
+            };
+
+                byte[] Extra22 = new byte[32]
```

Comment:
Formatting is far from perfect eg. here, but I have spent considerable time with regexes, fixers and formatters and do not believe further effort is worthwhile unless there is something egregious left, which I doubt.